src/Protocols/Wishbone.hs

```diff
@@ -62,14 +63,48 @@ data WishboneM2S addressWidth selWidth dat = WishboneM2S
     --   signals.
     burstTypeExtension :: "BTE" ::: BurstTypeExtension
   }
-  deriving (NFData, C.Generic, C.NFDataX, C.ShowX, Eq, C.BitPack)
+  deriving (NFData, C.Generic, C.NFDataX, Eq, C.BitPack)
 
--- M2S signals can contain undefined values, hence 'Show' is implemented through 'ShowX'
+instance
+  (C.ShowX dat, C.KnownNat addressWidth, C.KnownNat selWidth) =>
+  C.ShowX (WishboneM2S addressWidth selWidth dat)
+  where
+    showX = show
+
+-- Compact printing for M2S values. This handles undefined values in the
+-- structure too.
 instance
   (C.ShowX dat, C.KnownNat addressWidth, C.KnownNat selWidth) =>
   Show (WishboneM2S addressWidth selWidth dat)
   where
-  show = C.showX
+
+  show WishboneM2S{..} =
+    "WishboneM2S [ " <>
+    prefix busCycle <> "CYC " <>
+    prefix strobe <> "STB " <>
+    prefix writeEnable <> "WE, " <>
+    "ADR = " <> C.showX addr <> ", " <>
+    "DAT = " <> C.showX writeData <> ", " <>
+    "SEL = " <> C.showX busSelect <> ", " <>
+    "CTE = " <> cycle' <> ", " <>
+    "BTE = " <> burst <>
+    " ]"
```

Comment:
I could use `{}` over `[]` but I thought that it might be good to make it visible that it's not the default `Show` instance for records (which is a lot more verbose and makes debugging a bit more lengthy).

But I can use `{}` too, it's no issue!

Reply:
alright, seems good!